master/usage/encrypt-comms.md

```diff
@@ -0,0 +1,125 @@
+
+If you are using the etcd datastore, we recommend enabling mutual TLS authentication on
+its connections as follows.
+
+- [Configure etcd](https://coreos.com/etcd/docs/latest/op-guide/security.html) to encrypt its
+  communications with TLS and require clients to present certificates signed by the etcd certificate
+  authority.
+
+- Configure each {{site.prodname}} component to verify the etcd server's identity and to present
+  a certificate to the etcd server that is signed by the etcd certificate authority.
+  - [{{site.nodecontainer}}](../reference/node/configuration)
+  - [`calicoctl`](./calicoctl/configure/etcd)
+  - [CNI plugin](../reference/cni-plugin/configuration#etcd-location) (Kubernetes and OpenShift only)
+  - [Felix](../reference/felix/configuration#etcdv3-datastore-configuration) (on [bare metal hosts](../getting-started/bare-metal/installation/))
+  - [Typha](../reference/typa/configuration#etcdv3-datastore-configuration) (often deployed in
+    larger Kubernetes deployments)
+  - [Neutron or ML2 driver](https://docs.projectcalico.org/master/getting-started/openstack/) (OpenStack only)
+  - [DHCP agent](https://docs.projectcalico.org/master/getting-started/openstack/) (OpenStack only)
+
+### Connections from {{site.prodname}} components to kube-apiserver (Kubernetes and OpenShift)
+
+We recommend enabling TLS on kube-apiserver, as well as the client certificate and JSON web token (JWT)
+authentication modules. This ensures that all of its communications with {{site.prodname}} components occur
+over TLS 1.2. The {{site.prodname}} components present either an X.509 certificate
+or a JWT to kube-apiserver so that kube-apiserver can verify their identities.
+
+### Connections from Felix to Typha (Kubernetes)
+
+The communication between Felix and Typha instances can be secured with TLS.
+This protects against:
+
+-  A rogue or impostor Typha sending incorrect endpoint or policy information
+   to the Felix instances that connect to it
+
+-  A rogue or impostor Felix wrongly receiving privileged information about the
+   overall cluster from the Typha that it connects to
+
+-  Any other process discovering privileged information about the overall
+   cluster by eavesdropping on the communication channel
+
+To use TLS, each Typha and Felix instance must have a certificate and key pair signed by
+a trusted CA. Typha and Felix then only accept TLS connections, and require each
+connecting client to present a certificate that is signed by a trusted CA and
+has an expected identity in its Common Name or URI SAN field.
+
+-  For a [SPIFFE](https://github.com/spiffe/spiffe)-compliant deployment you
+   should configure `ClientURISAN` (on Typha) and `TyphaURISAN` (on Felix) with an
+   [SPIFFE Identity](https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md#2-spiffe-identity).
+   In addition, you should provision Typha and Felix with client certificates
+   that have the same identity in their URI SAN field.
```

Comment:
OK, after more reflection, I've realized that some of this doc is reference - which means that it should authoritatively and comprehensively state what the parameters do and what your options are for using them - and some of it is recommendation/usage/howto - which means that it's a recommended example of what to do.

I think it's actually a bit confusing to try to coalesce those, so I would now recommend:

- Reinstate the reference material on the Felix and Typha configuration pages; which means:
  - the `_includes/master/felix-typha-tls-intro.md` include
  - the "To use TLS..." and "If both those parameters are set..." paras.

- Put just the felix-typha-tls-howto.md material (which is a howto/recommendation) here.

WDYT?

Reply:
@neiljerram The reference material is supposed to be in the reference section (I have reinstated it in a previous commit) and the conceptual, how-to information is here. It is possible that procedural part of this could be further improved but it may be better to do that in a subsequent PR. I have made some edits to the intro part of this, let me know if it reads any better.